File: src/vec2/vec2.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__New = ____lualib.__TS__New
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["7"] = 1,["8"] = 1,["9"] = 1,["10"] = 6,["11"] = 7,["12"] = 8,["13"] = 6,["14"] = 16,["15"] = 17,["16"] = 18,["17"] = 19,["18"] = 16,["19"] = 27,["20"] = 28,["21"] = 29,["22"] = 30,["23"] = 27,["24"] = 38,["25"] = 39,["26"] = 40,["27"] = 41,["28"] = 38,["29"] = 49,["30"] = 50,["31"] = 51,["32"] = 52,["33"] = 49,["34"] = 60,["35"] = 61,["36"] = 62,["37"] = 63,["38"] = 60,["39"] = 70,["40"] = 71,["41"] = 70});
local ____exports = {}
____exports.Vec2 = __TS__Class()
local Vec2 = ____exports.Vec2
Vec2.name = "Vec2"
function Vec2.prototype.____constructor(self, x, y)
    self.x = x
    self.y = y
end
function Vec2.prototype.set(self, other)
    self.x = other.x
    self.y = other.y
    return self
end
function Vec2.prototype.add(self, other)
    self.x = self.x + other.x
    self.y = self.y + other.y
    return self
end
function Vec2.prototype.sub(self, other)
    self.x = self.x - other.x
    self.y = self.y - other.y
    return self
end
function Vec2.prototype.mul(self, other)
    self.x = self.x * other.x
    self.y = self.y * other.y
    return self
end
function Vec2.prototype.div(self, other)
    self.x = self.x / other.x
    self.y = self.y / other.y
    return self
end
function Vec2.prototype.clone(self)
    return __TS__New(____exports.Vec2, self.x, self.y)
end
return ____exports

File: src/vec2/vec2.ts
export class Vec2 {

  x: number;
  y: number;

  constructor(x: number, y: number) {
    this.x = x;
    this.y = y;
  }

  /**
   * Sets the original Vec2 to the other Vec2.
   * @param other Other Vec2.
   * @returns Original Vec2.
   */
  set(other: Vec2): Vec2 {
    this.x = other.x;
    this.y = other.y;
    return this;
  }

  /**
   * Addition between two Vec2. Mutates the original object.
   * @param other Other Vec2.
   * @returns Original Vec2.
   */
  add(other: Vec2): Vec2 {
    this.x += other.x;
    this.y += other.y;
    return this;
  }

  /**
   * Subtraction between two Vec2. Mutates the original object.
   * @param other Other Vec2.
   * @returns Original Vec2.
   */
  sub(other: Vec2): Vec2 {
    this.x -= other.x;
    this.y -= other.y;
    return this;
  }

  /**
   * Multiplication between two Vec2. Mutates the original object.
   * @param other Other Vec2.
   * @returns Original Vec2.
   */
  mul(other: Vec2): Vec2 {
    this.x *= other.x;
    this.y *= other.y;
    return this;
  }

  /**
   * Division between two Vec2. Mutates the original object.
   * @param other Other Vec2.
   * @returns Original Vec2.
   */
  div(other: Vec2): Vec2 {
    this.x /= other.x;
    this.y /= other.y;
    return this;
  }

  /**
   * Clone a Vec2.
   * @returns A clone of this Vec2.
   */
  clone(): Vec2 {
    return new Vec2(this.x, this.y);
  }
}